Add compression and decompression

hatchan opened the pull request:
Allow incoming request to be compressed using gzip or br and allow outgoing responses to be compressed using gzip or br (depending on the requested accept-encoding)

Resolves: FP-3985

brettimus commented:
there is already a changelog for mondrian 0.4.0, so i don't think i need to update the changelog!

hatchan committed:
Move axum to workspace

Make sure that axum features are only using what they require for the respective module

hatchan committed:
Add compression and decompression

Allow incoming request to be compressed using gzip or br and allow outgoing responses to be compressed using gzip or br (depending on the requested accept-encoding)
